src/Payments.sol

```diff
@@ -447,28 +456,49 @@ contract Payments is
         Account storage payee = accounts[rail.token][rail.to];
 
         uint256 oldRate = rail.paymentRate;
+        bool isTerminated = isRailTerminated(rail);
 
         // Settle the payer's lockup to account for elapsed time
         uint256 lockupSettledUpto = settleAccountLockup(payer);
 
         // Validate rate changes based on rail state and account lockup
-        validateRateChangeRequirements(
-            rail,
-            payer,
-            oldRate,
-            newRate,
-            lockupSettledUpto,
-            oneTimePayment
-        );
+        if (isTerminated) {
```

Comment:
I really want to see this change in a separate PR where I only have to look at changes related to this and I know what exactly we're trying to achieve. Otherwise, it's just moving code around and I have no idea what's going on.